tests/landsat/test_main.py

```diff
@@ -73,6 +83,171 @@ def test_qualifies_for_processing():
+                'autorift_parameters/v001/autorift_landice_0120m.shp',
+                'publish_bucket': '""',
+            },
+            'credit_cost': 1,
+        }
+    )
+    job2 = sdk.jobs.Job.from_dict(
+        {
+            'job_id': '4eea15af-167a-43b3-b292-aee55b3e893e',
+            'job_type': 'AUTORIFT',
+            'request_time': '2024-04-09T18:15:06+00:00',
+            'status_code': 'PENDING',
+            'user_id': 'cirrusasf',
+            'name': 'LC08_L1TP_138041_20240128_20240207_02_T1',
+            'job_parameters': {
+                'granules': ['LC08_L1TP_138041_20240128_20240207_02_T1', 'LC08_L1TP_138041_20231227_20240104_02_T1'],
+                'parameter_file': '/vsicurl/http://its-live-data.s3.amazonaws.com/'
+                'autorift_parameters/v001/autorift_landice_0120m.shp',
+                'publish_bucket': '""',
+            },
+            'credit_cost': 1,
+        }
+    )
+
+    jobs_expect = sdk.jobs.Batch([job1, job2])
+    return jobs_expect
+
+
+def test_get_stac_item():
+    scene = 'LC08_L1TP_138041_20240128_20240207_02_T1'
+    expect_item = get_expect_item()
+
+    main.LANDSAT_CATALOG = MagicMock()
+    main.LANDSAT_CATALOG.get_collection().get_item.return_value = expect_item
+
+    item = main._get_stac_item(scene)
+
+    assert item.collection_id == expect_item.collection_id
+    assert item.properties['instruments'] == expect_item.properties['instruments']
+    assert item.properties['landsat:wrs_path'] == expect_item.properties['landsat:wrs_path']
+    assert item.properties['landsat:wrs_row'] == expect_item.properties['landsat:wrs_row']
+    assert item.properties['view:off_nadir'] == expect_item.properties['view:off_nadir']
+    assert item.properties['landsat:cloud_cover_land'] == expect_item.properties['landsat:cloud_cover_land']
+    assert item.properties['landsat:collection_category'] == expect_item.properties['landsat:collection_category']
+
+
+def test_get_landsat_pairs_for_reference_scene():
+    main.LANDSAT_CATALOG = MagicMock()
+    reference_item = get_expect_item()
+    results_item_collection = pystac.item_collection.ItemCollection.from_file(
+        'tests/data/scene1_return_itemcollection.json'
+    )
+    data_gen = (y for y in [results_item_collection])
+    main.LANDSAT_CATALOG.search().pages.return_value = data_gen
+
+    df = main.get_landsat_pairs_for_reference_scene(reference_item)
+
+    assert (df['landsat:wrs_path'] == reference_item.properties['landsat:wrs_path']).all()
+    assert (df['landsat:wrs_row'] == reference_item.properties['landsat:wrs_row']).all()
+    assert (df['view:off_nadir'] == reference_item.properties['view:off_nadir']).all()
+    assert (df['instruments'].apply(lambda x: ''.join(x)) == ''.join(reference_item.properties['instruments'])).all()
+    assert (df['reference'] == reference_item.id).all()
+
+
+def test_deduplicate_hyp3_pairs():
+    pairs = gpd.read_parquet('tests/data/scene1_pair.parquet')
```

Comment:
If you do the `SAMPLE_PAIRS=gpd.read_parquet(...)` as a constant (global variable) then you can add it as an argument here and in `test_submit_pairs_for_processing` like this:
```suggestion
def test_deduplicate_hyp3_pairs(pairs=SAMPLE_PAIRS):
```